src/module.cpp

```diff
@@ -732,6 +751,194 @@ void module::finalize(std::vector<context>& contexts)
                   << std::endl;
 }
 
+std::unordered_map<instruction_ref, instruction_ref>
+module::get_ins_param_map(const std::vector<instruction_ref>& inputs, bool reverse) const
+{
+    std::unordered_map<instruction_ref, instruction_ref> result;
+    auto params = this->get_parameters();
+    assert(params.size() == inputs.size());
+    sort_params(params);
+    if(reverse)
+    {
+        std::transform(
+            params.begin(),
+            params.end(),
+            inputs.begin(),
+            std::inserter(result, result.end()),
+            [&](instruction_ref param, auto input) { return std::make_pair(param, input); });
+    }
+    else
+    {
+        std::transform(
+            params.begin(),
+            params.end(),
+            inputs.begin(),
+            std::inserter(result, result.end()),
+            [&](instruction_ref param, auto input) { return std::make_pair(input, param); });
+    }
```

Comment:
The parameters are flipped in the `make_pair` call. The reverse will reverse the lookup so instead of `ins -> param_ins` it will be `param_ins -> ins`.